### PR TITLE
various: fix and improve several things

### PR DESCRIPTION
**Pull request by rerowep**
pilot: several improvements

    * Implements lazy reading of xml files.
    * Adds xml record extraction.
    * Adds operation_logs pid dependencies checking.
    * Adds paramter for error records saving during record creation.

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by lauren-d — APPROVED** [review]

File: rero_ils/modules/cli.py [hunk 1]
```diff
@@ -372,6 +384,9 @@ def create(infile, append, reindex, dbcommit, commit, verbose, debug, schema,
     click.echo('DB commit: {count}'.format(count=count))
     db.session.commit()
 
+    if save_errors:
+        error_file.write(']')
```
Comment (zannkukai) [inline rero_ils/modules/cli.py:388]: missing a `error_file.close()` at the end of the method. Previously it was made by the `with` statement, now i think you need do it yourself.

**Review by zannkukai — CHANGES_REQUESTED** [review]

File: rero_ils/modules/cli.py (hunk 1 above)
Comment (rerowep) [reply rero_ils/modules/cli.py:389]: ok

**Review by zannkukai — APPROVED** [review]

**Review by jma — APPROVED** [review]

**Review by iGormilhit — COMMENTED** [review]
I don't know how you can improve the commit message title, but:

- pilot is not a RERO ILS component.
- the second part (after the semi-colon) should be a sentence, begining with a verb. Something like 'Fixes several bugs' is better.

**Review by BadrAly — APPROVED** [review]

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message proposition:

```
various: fix and improve several things

* Uses an existing DB for deployment.
* Implements lazy reading of XML files.
* Adds XML record extraction.
* Checks operation_logs PID dependencies.
* Adds parameter to log errors on records creation.
* Fixes item creation with holdings without patterns.
* Fixes editor options for docmaintype_audio subtypes. Closes #1712.
```

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

**Pull request merged**